src/routes/committee/setlist/+page.svelte

```diff
@@ -167,118 +115,47 @@
 						size="small"
 						iconDescription="Add musician"
 						icon={UserFollow}
-						on:click={() => {
-							selectedSong = i
-							openAddMusician = true
-						}}
+						on:click={() => openModal({ state: 'addMusician', song })}
 					/>
 					<Button
 						kind="ghost"
 						size="small"
 						iconDescription="Edit"
 						icon={Edit}
-						on:click={() => {
-							selectedSong = i
-							openEditSong = true
-						}}
+						on:click={() => openModal({ state: 'editSong', song })}
 					/>
 					<Button
 						kind="danger-tertiary"
 						size="small"
 						iconDescription="Delete"
 						icon={MusicRemove}
-						on:click={() => {
-							selectedSong = i
-							openDelSong = true
-						}}
+						on:click={() => openModal({ state: 'deleteSong', song })}
 					/>
 				</StructuredListCell>
 			</StructuredListRow>
 		{/each}
 	</ScrollableList>
 </Grid>
 
-<AddSongModal bind:openAddSong />
-
-{#if data.songs[selectedSong]}
-	<EditSongModal bind:openEditSong song={data.songs[selectedSong]} />
+{#if modalState.state === 'addSong'}
+	<AddSongModal {...modalState} onClose={() => closeModal()} />
+{:else if modalState.state === 'editSong'}
+	<EditSongModal {...modalState} onClose={() => closeModal()} />
+{:else if modalState.state === 'deleteSong'}
+	<DeleteSongModal {...modalState} onClose={() => closeModal()} />
+{:else if modalState.state === 'addMusician'}
+	<AddParticipantToSongModal {...modalState} users={data.users} onClose={() => closeModal()} />
+{:else if modalState.state === 'deleteMusician'}
+	<RemoveParticipantFromSongModal {...modalState} onClose={() => closeModal()} />
```

Comment:
Ahh Tim my boy, ik zie het al. Je maakt het moeilijker voor jezelf door de props te destructuren. Typescript kan idd niet inferren dat props B en C bestaan als prop A een bepaalde waarde heeft, maar als je nou zorgt dat de modal niet 4 props neemt maar 1 prop dat een object is bestaande uit die 4 dingen, dan gaat dat qua typing goed. Ik ga niet voor je uitzoeken wat je daardoor allemaal netter kunt doen in deze merge, maar ik ga er vanuit dat het aardig wat is

Reply:
Denk ik, misschien klopt wat ik zeg niet helemaal omdat die #if van svelte nog niet perfect samengaat met typescript, zal zo eens kijken of dat geupdate is ondertussen. Maar als dat het geval is kun je wat mij betreft ook gewoon casten

Reply:
Maar dan nog wel conditionally renderen?

Reply:
Want anders weet svelte/typescript (dunno which one) nooit welke props er zijn

Reply:
Volgens mij is het destructuren `{...modalState}` niet een probleem?

Reply:
> Typescript kan idd niet inferren dat props B en C bestaan als prop A een bepaalde waarde heeft

It does though...?